hibernate-core/src/main/java/org/hibernate/engine/jdbc/batch/internal/BatchBuilderImpl.java

```diff
@@ -66,9 +66,26 @@ public Batch buildBatch(BatchKey key, JdbcCoordinator jdbcCoordinator) {
 		final int jdbcBatchSizeToUse = sessionJdbcBatchSize == null ?
 				this.jdbcBatchSize :
 				sessionJdbcBatchSize;
-		return jdbcBatchSizeToUse > 1
-				? new BatchingBatch( key, jdbcCoordinator, jdbcBatchSizeToUse )
-				: new NonBatchingBatch( key, jdbcCoordinator );
+/**
+IMO, No need to use NoBatchingBatch class, because
+1.It duplicates the if/else condition for example from AbstractEntityPersister:3163 and
+2.useBatch - variable is kind of responsible to use batch or not - keep single responsibility
+jdbcBatchSizeToUse > 1 - condition can be checked in AbstractEntityPersister and adjust the useBatch variable
```

Comment:
What if the dev provided a batch size of 1? I don't really see a need to change this right now.

Reply:
The idea is that i'd preffer `BatchBuilderImpl` to have the `buildBatch` method passing and the jdbcBatchSize. 
`public Batch buildBatch(BatchKey key, JdbcCoordinator jdbcCoordinator,int jdbcSize){}`
so that here we only create the batch. The check itself of creating a batch or not is moved on abstract layer in `AbstractEntityPersister`. Besides this , i noticed that the `NonBatchingBatch.class` kind of duplicates the logic from `AbstractEntityPersister.check()` method, that's it why makes no sense to use NonBatchingBatch class when we cant batch